multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedMultimapSortedSetCache.java

```diff
@@ -329,4 +338,68 @@ public CompletionStage<Collection<SortedSetBucket.ScoredValue<V>>> inter(K key,
       SortedSetBucket.AggregateFunction agg = aggFunction == null ? SortedSetBucket.AggregateFunction.SUM : aggFunction;
       return readWriteMap.eval(key, new SortedSetAggregateFunction(INTER, scoredValues, weight, agg));
    }
+   /**
+    * Removes the given elements from the sorted set, if such exist.
+    * @param key, the name of the sorted set
+    * @param members, members to be removed
+    * @return removed members count
+    */
+   public CompletionStage<Long> removeAll(K key, List<V> members) {
+      requireNonNull(key, ERR_KEY_CAN_T_BE_NULL);
+      requireNonNull(members, ERR_MEMBERS_CAN_T_BE_NULL);
+      return readWriteMap.eval(key, new RemoveManyFunction<>(members, OTHER));
+   }
+
+   /**
+    * Removes range from index from to index to.
+    *
+    * @param key, the name of the sorted set
+    * @param min, index from
+    * @param max, index to
+    * @return long representing the number of members removed
+    */
+   public CompletionStage<Long> removeAll(K key, Long min, Long max) {
+      requireNonNull(min, ERR_MIN_CAN_T_BE_NULL);
+      requireNonNull(max, ERR_MAX_CAN_T_BE_NULL);
+      return removeAll(key, min, false,  max, false, INDEX);
+   }
+
+   /**
+    * Removes elements from a range of scores.
+    *
+    * @param key, the sorted set name
+    * @param min, smallest score value to be removed
+    * @param includeMin, indicates if the min score is included in the remove range
+    * @param max, greatest score value to be removed
+    * @param includeMax, indicates if the max score is included in the remove range
+    *
+    * @return the number of removed elements
+    */
+   public CompletionStage<Long> removeAll(K key, Double min, boolean includeMin, Double max, boolean includeMax) {
```

Comment:
In regards to non-null arguments. We could change these to a double primitive.

Reply:
no, not possible for what I've explained above. is the same with zrange or zcount etc.